generators/app/templates/src/containers/Profile/components/Image/image.component.js

```diff
@@ -0,0 +1,94 @@
+import React, { useState, useEffect } from 'react';
+import data from "@solid/query-ldflex";
+import { Uploader, useLiveUpdate } from '@inrupt/solid-react-components';
+import { ImageProfile } from '@components';
+
+type Props = {
+  webId: String,
+  toastManager: String
+};
+
+
+export const Image = ({ webId, toastManager, defaultProfilePhoto }: Props) => {
+  const [image, setImage] = useState('');
+
+  const latestUpdate = useLiveUpdate();
+
+  useEffect( () => {
+    fetchPhoto();
+  }, [webId, latestUpdate]);
+  /**
+   * Fetch profile photo from card
+   */
+  const fetchPhoto = async () => {
+    try {
+      if (webId) {
+        // We are fetching profile card document
+        const user = data[webId];
+        // We access to document node using a node name
+        let image = await user.image;
+        // If image is not present on card we try with hasPhoto
+        if (!image) {
+          /**
+           * hasPhoto is a new context that ldflex doesn't having
+           * we need to add it manually.
+           * if you want to know more about context please go to:
+           * https://github.com/digitalbazaar/jsonld.js
+           */
+          image = await user.vcard_hasPhoto;
+        }
+
+        setImage(image && image.value);
+      }
+    } catch (error) {
+      toastManager.add (['Error', error.message], {
+        appearance: 'error',
+      });
+    }
+  };
+  /**
+   * updatedPhoto will update the photo url on vcard file
+   * this function will check if user has image or hasPhoto node if not
+   * will just update it, the idea is use image instead of hasPhoto
+   * @params{String} uri photo url
+   */
+  const updatePhoto = async (uri: String) => {
+    try {
+      const {user} = data;
+      image
+        ? await user.image.set (uri)
+        : await user.image.add (uri);
+
+      toastManager.add (['', 'Profile Image was updated'], {
+        appearance: 'success',
+      });
+    } catch (error) {
+      toastManager.add (['Error', error.message], {
+        appearance: 'error',
+      });
+    }
```

Comment:
Your `toastManager` code takes up a lot of lines.
Consider creating shortcuts like `toastManager.success` and `toastManager.error`.

Reply:
You can even avoid the `catch` if you write a helper like
```
toastManager.report(async () => {
}, "Profile image update");
```

with

```
toastManager.report = async function(action, successMsg) {
   try { await action(); }
   catch (error) { toastManager.error(error); return; }
   toastManager.success(successMsg);
}
```

Reply:
We want to remove this toastsManager and create a new one to avoid the use of consumer in all places, so I don't know if we need to do this refactor since we will remove it.